src/master/master.ts

```diff
@@ -114,9 +78,31 @@ export type TransportData =
       args: [string, ChatMessage];
     };
 
+export type IntermediateTransportData =
```

Comment:
There’s quite a bit of duplication between this and the `TransportData` type. Could we extract the types shared between both?

```ts
type CommonTransportData =  { /* ... */ } | { /* ... */ };
type TransportData = CommonTransportData | { /* ... */ };
type IntermediateTransportData = CommonTransportData | { /* ... */ };
```

We could even break out the related data types if you like:

```ts
import type { List } from 'ts-toolbelt';

// ...

type IntermediateUpdateData = {
  type: 'update';
  args: [string, State];
};

type UpdateData = {
  type: IntermediateUpdateData['type'];
  args: List.Append<IntermediateUpdateData['args'], LogEntry[]>;
};
```

(With Typescript 4 we wouldn’t need the `ts-toolbelt` helper and could write `args: [...IntermediateUpdateData['args'], LogEntry[]]`, but I don’t remember if  updating caused problems with Rollup.)

Reply:
I went with your first suggestion, I am not very familiar with the ts-toolbelt and I felt like it could make things harder to read (at cost of some repetition).